assets/src/utils/dagre.ts

```diff
@@ -68,3 +71,36 @@ export function createDagre(
     },
   });
 }
+
+const graphInteractionEvents = ["click", "dragstart", "touchstart"];
+
+export function graphInteractionListener(
+  graph: Graph,
+  handler: () => void
+): () => void {
+  return function() {
+    graphInteractionEvents.forEach((event) => graph.on(event, () => {
+      handler();
+      graphInteractionEvents.forEach((event) => graph.off(event));
+    }));
+  };
+}
+
+export function nodeIdsDifferent(
+  nodes1: NodeConfig[],
+  nodes2: NodeConfig[]
+): boolean {
+  if (nodes1.length !== nodes2.length) return true;
+
+  const [bigger, smaller] =
+    nodes1.length > nodes2.length
+      ? [nodes1, nodes2]
+      : [nodes2, nodes1];
+
+  const idSet = new Set(bigger.map((node) => node.id));
+  for (const node of smaller) {
```

Comment:
use `.some()` function of js arrays, it is more elegant than for loops.

Reply:
Didn't know that one, refactored in 3aa9e1c